x-pack/plugins/uptime_monitoring/public/components/functional/snapshot_histogram.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore Missing typings for series charts
+import { EuiHistogramSeries, EuiSeriesChart, EuiSeriesChartUtils } from '@elastic/eui';
+import React from 'react';
+import { HistogramDataPoint, HistogramSeries } from '../../../common/graphql/types';
+
+const formatHistogramData = (histogram: HistogramSeries[]) => {
```

Comment:
It seems it might be good to export this or have it in its own file to unit test?

Reply:
++ - will move to a dedicated file. In the future I'll probably move that code to the server, I don't necessarily like the idea of the client formatting data anyway. The reason this is here in the first place is because I misunderstood how the series chart components worked and that I couldn't easily create a swimlane, so rather than update the schema, resolvers, and ES I made this hack.

https://github.com/elastic/kibana/pull/27279/commits/8e6fc41262c48269335bd464bf56de1f16ce8483